frontend/packages/pipelines-plugin/src/components/repository/repository-hooks.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { k8sGet } from '@console/internal/module/k8s';
+import { PipelineRunModel } from '../../models';
+import { getLatestRun } from '../../utils/pipeline-augment';
+
+const useGetPipelineRuns = (pipelineRunsName: string[], namespace: string) => {
+  const [pipelineRuns, setPipelineRuns] = React.useState([]);
+  React.useEffect(() => {
+    if (pipelineRunsName.length > 0) {
+      Promise.all(
+        pipelineRunsName.map((pipelineRunName) => {
+          return k8sGet(PipelineRunModel, pipelineRunName, namespace).then((plr) => plr);
+        }),
+      )
+        .then((data) => {
+          setPipelineRuns(data);
+        })
+        .catch((err) => err);
+    }
+  }, [namespace, pipelineRunsName]);
```

Comment:
Need a return function that will suppress the return if the hook unloads in the middle of the API call. [Something like this](https://github.com/openshift/console/blob/master/frontend/packages/pipelines-plugin/src/components/catalog/hooks/useApiResponse.ts#L35-L37).

Reply:
Added